launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -182,6 +182,9 @@ List<String> buildClassPath(String appClassPath) throws IOException {
           if (project.equals("sql/connect/server") || project.equals("sql/connect/common")) {
             continue;
           }
+          if (isRemote && "1".equals(getenv("SPARK_SCALA_SHELL")) && project.equals("sql/core")) {
```

Comment:
cc @HyukjinKwon FYI

Is this check condition correct?

Reply:
thank you for fixing this quickly